### PR TITLE
HPCC-20753 Enable persistent configuration for httpclient

### PR DESCRIPTION
**Pull request by mayx**
- Utitlize the 2 config entries maxPersistentIdleTime and
  maxPersistentRequests in httpclient

Signed-off-by: mayx <yanrui.ma@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [x] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [x] Scalability
  - [x] Performance
  - [x] Security
  - [x] Thread-safety
  - [x] Premature optimization
  - [x] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [x] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by mayx** [comment]
@afishbeck please review...

File: esp/bindings/http/client/httpclient.cpp [hunk 1]
```diff
@@ -50,7 +50,31 @@ CHttpClientContext::CHttpClientContext(IPropertyTree* config) : m_config(config)
...
 void CHttpClientContext::initPersistentHandler()
 {
-    m_persistentHandler.setown(createPersistentHandler(nullptr, DEFAULT_MAX_PERSISTENT_IDLE_TIME, DEFAULT_MAX_PERSISTENT_REQUESTS, static_cast<PersistentLogLevel>(getEspLogLevel()), true));
+    IEspServer* server = queryEspServer();
+    IPropertyTree* proc_cfg = nullptr;
+    if (server)
+        proc_cfg = server->queryProcConfig();
+    const char* idleTimeStr = nullptr;
+    const char* maxReqsStr = nullptr;
+    if (proc_cfg != nullptr)
```
Comment (afishbeck) [inline esp/bindings/http/client/httpclient.cpp]: Could just use?:
int maxIdleTime = DEFAULT_MAX_PERSISTENT_IDLE_TIME;
int maxReqs = DEFAULT_MAX_PERSISTENT_REQUESTS;
if (proc_cfg)
{
    maxIdleTime = proc_cfg->queryPropInt("@maxPersistentIdleTime", maxIdleTime);
    maxReqs = proc_cfg->queryPropInt("@maxPersistentRequests", maxReqs);
}

**Review by afishbeck — COMMENTED** [review]
@mayx looks good, but one comment.

File: esp/bindings/http/client/httpclient.cpp (hunk 1 above)
Comment (mayx) [reply esp/bindings/http/client/httpclient.cpp]: Good point, let me make the change.

**Comment by mayx** [comment]
@afishbeck change uploaded, please review again...

**Review by afishbeck — APPROVED** [review]
@mayx looks good.

**Comment by richardkchapman** [comment]
Should this be targetting 7.2.x?

**Comment by afishbeck** [comment]
This provides a configuration workaround for a major performance issue that Yanrui still has to investigate. @richardkchapman

**Comment by mayx** [comment]
@richardkchapman yes I think we should put it in both 7.0 and 7.2.x

**Pull request merged**